src/engine/idTable/CompressedExternalIdTable.h

```diff
@@ -378,6 +378,10 @@ class CompressedExternalIdTableBase {
       compressAndWriteFuture_.get();
     }
     if (block.empty()) {
+      // BUGFIX: It's important to set the future, even if it does nothing
+      // (otherwise the assertion in `transformAndPushLastBlock` fails).
+      compressAndWriteFuture_ = std::async(
+          std::launch::deferred, [this]() { });
```

Comment:
I think I have found a possible issue with your "fix". If ALL the blocks are empty, then we still run into trouble (because `numBlocksPushed == 0`, but the future is now valid (because you have made it valid here). So maybe the fix is to have the following logic:

```
if (block.empty()) {
  if (numBlocksPushed_ > 0) { engageTheFutureWithDummy();}
  return;
}
```

Try it out and let me know if that fixes your problem.